internal/resolution/lockfile/npm.go

```diff
@@ -19,14 +19,17 @@ type NpmReadWriter struct{}
 
 func (NpmReadWriter) System() resolve.System { return resolve.NPM }
 
+type npmDependencyVersionSpec struct {
+	Version string
+	DepType dep.Type
+}
+
 type npmNodeModule struct {
-	NodeID       resolve.NodeID
-	Parent       *npmNodeModule
-	Children     map[string]*npmNodeModule // keyed on package name
-	Deps         map[string]string
-	OptionalDeps map[string]string
-	DevDeps      map[string]string // dev dependencies are also included in Deps
```

Comment:
can you explain a  bit more why we don't need `DevDeps` and `OptionalDeps` here anymore?

Reply:
`Deps` now also contains the `dep.Type`, which has whether it's optional or dev (or peer) in it.